src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -35,7 +35,7 @@ public CsProjGenerator(string targetFrameworkMoniker, string cliPath, string pac
 
         protected override string GetBuildArtifactsDirectoryPath(BuildPartition buildPartition, string programName)
         {
-            string assemblyLocation = buildPartition.AssemblyLocation;
+            string assemblyLocation = buildPartition.RepresentativeBenchmarkCase.Descriptor.Type.Assembly.Location;
```

Comment:
why are you not using `buildPartition.AssemblyLocation` here?

Reply:
This is a separate mechanism for “assembly loaded from stream”. I didn’t wanted to change/refactor it too much, as here we expect Location to return null for LoadFromStream case and test fails if Location is non-null.